openpype/hosts/max/plugins/publish/collect_review.py

```diff
@@ -24,25 +27,26 @@ def process(self, instance):
         for node in nodes:
             if rt.classOf(node) in rt.Camera.classes:
                 camera_name = node.name
-                focal_length = node.fov
-
+                if rt.isProperty(node, "fov"):
+                    focal_length = node.fov
+                else:
+                    raise KnownPublishError(
+                        "Invalid object found in 'Review' container."
+                        " Only native max Camera supported"
+                    )
```

Comment:
> The code is a bit weird - because if you put in two valid cameras into the instance, no errors will occur and it will randomly use one of the two without warning.
> 
> ```python
> def is_camera(node):
>     is_camera_class = rt.classOf(node) in rt.Camera.classes
>     return is_camera_class and rt.isProperty(node, "fov")
> 
> # Use first camera in instance
> cameras = [node for node in nodes if is_camera(node)]
> if cameras:
>     if len(cameras) > 1:
>         self.log.warning(
>              "Found more than one camera in instance, using first "
>              f"one found: {cameras[0]}"
>         )
>     camera = cameras[0]
>     camera_name = camera.name
>     focal_length = camera.fov
> ```
> 
> How about this?
> 
> Also, out of curiosity what types of Camera classes exist in Max that do **not** support `fov`? Orthographic ones? If so I'd expect to also be able to review those and if that's the case that `fov` either remains `None` or gets set to the orthographic width instead or just becomes 0 as if unknown.
> 
> Thoughts?

The imported ABC camera doesn't support the fov, they actually dont have the fov as the parameter where we can access them to get the info for focal length.

Reply:
Interesting - that'd be confusing I suppose.

Reply:
they have totally different attributes to be accessed but these do not include the focal length. Hopefully they would expose the data in the future.